sdks/python/apache_beam/transforms/core.py

```diff
@@ -2587,6 +2587,18 @@ def add_input_types(transform):
             "or CombineGlobally().as_singleton_view() to get the default "
             "output of the CombineFn if the input PCollection is empty.")
 
+      # return the error for this ill-defined streaming case
+      if not pcoll.is_bounded and not pcoll.windowing.is_default():
```

Comment:
Could you please add a docstring for `Windowing.is_default()` while at it? I wonder if you actually wanted to refer to `CombineGlobally.has_defaults` here...